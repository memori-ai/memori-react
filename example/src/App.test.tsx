import { describe, expect, it } from 'vitest';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

afterEach(() => {
  cleanup();
});

describe('Simple working test', () => {
  it('the title is visible', () => {
    render(<App />);
    expect(screen.getByText(/Hello Vite \+ React!/i)).toBeInTheDocument();
  });

  it('should increment count on click', async () => {
    render(<App />);
    userEvent.click(screen.getByRole('button'));
    expect(await screen.findByText(/count is: 1/im)).toBeInTheDocument();
  });

  it('uses flexbox in app header', async () => {
    render(<App />);
    const element = screen.getByRole('banner');
    expect(element.className).toEqual('App-header');
    expect(getComputedStyle(element).display).toEqual('flex');
  });
});

describe('Snapshot test', () => {
  it('renders App unchanged', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
