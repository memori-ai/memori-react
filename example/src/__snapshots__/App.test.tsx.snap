// Vitest Snapshot v1

exports[`Snapshot test > renders App unchanged 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <a
        href="https://vitejs.dev"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/vite.svg"
        />
      </a>
      <a
        href="https://reactjs.org"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react"
          src="/react.svg"
        />
      </a>
    </header>
    <main>
      <h1>
        Hello Vite + React!
      </h1>
      <div
        class="card"
      >
        <button>
          Increment
        </button>
        <p>
          Count is: 
          0
        </p>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR
        </p>
      </div>
    </main>
    <p
      class="read-the-docs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</div>
`;
